src/connector/src/sink/kafka.rs

```diff
@@ -228,7 +228,10 @@ pub struct KafkaConfig {
     pub primary_key: Option<String>,
 
     #[serde(flatten)]
-    pub rdkafka_properties: RdKafkaPropertiesProducer,
+    pub rdkafka_properties_common: RdKafkaPropertiesCommon,
```

Comment:
Are we assuming a "*PropertiesCommon" type will occur at most once in a WithOptions struct? Otherwise after flattened field names will conflict between multiple instances?

Reply:
Yes. All flatten fields (the "common" struct) are put in the top-level struct (thus only once). Children types don't have nested fields, and won't conflict.